caddyconfig/httpcaddyfile/serveroptions.go

```diff
@@ -217,6 +218,12 @@ func unmarshalCaddyfileServerOptions(d *caddyfile.Dispenser) (any, error) {
 				)
 				serverOpts.TrustedProxiesRaw = jsonSource
 
+			case "enable_strict_xff_parsing":
```

Comment:
I don't like this name because we may not actually be parsing XFF because of `ClientIPHeaders`. I think something like `trusted_proxies_strict` or `trusted_proxies_mode strict` (if we want to have more than one mode) might be more sensible; uses a similar prefix to the related option so the relationship between them is more obvious.

Reply:
great call. can you point me in the right direction of the Golang-way of introducing an option set? e.g. "strict", "left-most", etc. do you just do string matching in the `serveroptions.go` parser?

Reply:
Oh sorry, I saw your comment and meant to reply but got sidetracked.

Yeah we generally just do string matching for modes, with a `switch` statement etc. Sometimes we'll use a const in code. See the `auto_https` option for examples of modes.

Reply:
In https://adam-p.ca/blog/2022/03/x-forwarded-for/ – the article details only one scenario where you'd want left-most from a security perspective: if you have a non-conformant load balancer that appends XFF to the wrong side. Otherwise, right-most is the appropriate approach for security and correctness, and left-most is the appropriate approach for the rest.

Let me know what you think. Happy to change this to a switch statement if there could be more.